src/sx/src/keyframes.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+
+import { isBrowser } from '@adeira/js';
+import stringify from 'json-stable-stringify';
+
+import hash from './hashStyle';
+import type { AllCSSPropertyTypes } from './css-properties/__generated__/AllCSSPropertyTypes';
+import styleCollector from './StyleCollector';
+import transformStyleName from './transformStyleName';
+import transformStyleValue from './transformValue';
+import { injectCSS } from './injectRuntimeStyles';
+
+type KeyFrames = {|
+  +from?: AllCSSPropertyTypes,
+  +to?: AllCSSPropertyTypes,
+  +[number]: AllCSSPropertyTypes, // percentages
```

Comment:
I was thinking that maybe we should change it to string:

```suggestion
  +[string]: AllCSSPropertyTypes, // percentages
```

It's because keyframes can be more complex:

```css
@keyframes identifier {
  0% { top: 0; left: 0; }
  30% { top: 50px; }
  68%, 72% { left: 50px; }
  100% { top: 100px; left: 100%; }
}
```

See: https://developer.mozilla.org/en-US/docs/Web/CSS/@keyframes#When_properties_are_left_out_of_some_keyframes

Do you agree?

Reply:
Yeah, that makes sense 👍